### PR TITLE
Swallowing Objects

### PR DESCRIPTION
**Pull request by dragoslava**
Adds the ability to swallow smaller objects to hide in your stomache that can be coughed up/removed later on.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 1]
```diff
@@ -270,6 +270,38 @@
 			else if(prob(50))
 				to_chat(H,"<span class='warning'>Your mind buzzes...</span>")
 
+/mob/living/carbon/human/proc/swallowobject()
+	set name = "Swallow Object"
+	set desc = "Swallows a held object"
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
	set desc = "Swallows a held object."
```

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 2]
```diff
@@ -270,6 +270,38 @@
...
+	set category = "IC"
+
+	var/mob/living/carbon/human/H = src
+	var/obj/item/organ/ST = H.internal_organs_by_name["stomache"]
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
	var/obj/item/organ/ST = H.internal_organs_by_name["stomach"]
```

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 3]
```diff
@@ -270,6 +270,38 @@
...
+			to_chat(H, "<span class='notice'>You swallow the [I], but feel a intese stabbing pain!</span>")
+			I.forceMove(ST)
+			E.createwound(CUT, rand(30,50), 1)
+			for(var/datum/wound/W in E.wounds)
+				if(W.internal)
+					E.wounds += W
+
+	if(I.w_class <= 2)
+		if(do_after(H, 60, act_target = I))
+			to_chat(H, "<span class='notice'>This is too big to swallow</span>")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
			to_chat(H, span("notice","This is too big to swallow."))
```

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 4]
```diff
@@ -270,6 +270,38 @@
...
+		to_chat(H, "<span class='notice'>You can't swallow nothing!</span>")
+		return
+	if(I.w_class >= 2)
+		to_chat(H, "<span class='notice'>You begin to swallow the [I]</span>")
+		if(do_after(H, 40, act_target = I))
+			H.drop_from_inventory(I,H.loc)
+			I.forceMove(ST)
+			to_chat(H, "<span class='notice'>You swallow the [I]</span>")
+		if(I.sharp == 1)
+			to_chat(H, "<span class='notice'>You swallow the [I], but feel a intese stabbing pain!</span>")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
			to_chat(H, span("warning", "You swallow \the [I], but feel an intense stabbing pain!"))
```

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 5]
```diff
@@ -173,6 +173,7 @@
 		"kidneys" =  /obj/item/organ/kidneys,
 		"brain" =    /obj/item/organ/brain,
 		"appendix" = /obj/item/organ/appendix,
+		"stomache" = /obj/item/organ/stomache,
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/species/species.dm]:
```suggestion
		"stomach" = /obj/item/organ/stomach,
```

File: code/modules/mob/mob_helpers.dm [hunk 6]
```diff
@@ -723,7 +723,8 @@ proc/is_blind(A)
 
 /mob/living/carbon/proc/vomit()
 	var/canVomit = 0
-	var/mob/living/carbon/human/H
+	var/mob/living/carbon/human/H = src
+	var/obj/item/organ/ST = H.internal_organs_by_name["stomache"]
```
Comment (MarinaGryphon) [inline code/modules/mob/mob_helpers.dm]:
```suggestion
	var/obj/item/organ/ST = H.internal_organs_by_name["stomach"]
```

File: code/modules/mob/mob_helpers.dm [hunk 7]
```diff
@@ -740,13 +741,17 @@ proc/is_blind(A)
 		var/turf/location = loc
 		if (istype(location, /turf/simulated))
 			location.add_vomit_floor(src, 1)
+		if(ST.contents.len)
+			var/atom/movable/vomititem = pick(ST.contents)
+			vomititem.forceMove(src.loc)
+			src.visible_message("<span class='warning'>[src] spits up the[vomititem.name]</span>","<span class='warning'>You vomit a object up</span>")
```
Comment (MarinaGryphon) [inline code/modules/mob/mob_helpers.dm]:
```suggestion
			src.visible_message(span("warning", "[src] vomits up \the [vomititem]!"), span("warning", "You vomit something up!"))
```

File: code/modules/organs/organ_internal.dm [hunk 8]
```diff
@@ -15,6 +15,40 @@
 	robotic_name = "circulatory pump"
 	robotic_sprite = "heart-prosthetic"
 
+/obj/item/organ/stomache
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
/obj/item/organ/stomach
```

File: code/modules/organs/organ_internal.dm [hunk 9]
```diff
@@ -15,6 +15,40 @@
...
+	name = "stomache"
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
	name = "stomach"
```

File: code/modules/organs/organ_internal.dm [hunk 10]
```diff
@@ -15,6 +15,40 @@
...
+	icon_state = "stomache"
+	gender = PLURAL
+	organ_tag = "stomache"
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
	organ_tag = "stomach"
```

File: code/modules/organs/organ_internal.dm [hunk 11]
```diff
@@ -15,6 +15,40 @@
...
+	parent_organ = "groin"
+	robotic_name = "gas exchange system"
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]: This needs to be changed.

File: code/modules/organs/organ_internal.dm [hunk 12]
```diff
@@ -15,6 +15,40 @@
...
+	robotic_sprite = "heart-prosthetic"
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]: This also needs to be changed.

File: code/modules/organs/organ_internal.dm [hunk 13]
```diff
@@ -15,6 +15,40 @@
...
+	var/vomit_timer = FALSE
+
+/obj/item/organ/stomache/Initialize()
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
/obj/item/organ/stomach/Initialize()
```

File: code/modules/organs/organ_internal.dm [hunk 14]
```diff
@@ -15,6 +15,40 @@
...
+	START_PROCESSING(SSfast_process, src)
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]: Is there not a better way to do this? Such as checking if it's too full only when its fullness is modified?

File: code/modules/organs/organ_internal.dm [hunk 15]
```diff
@@ -15,6 +15,40 @@
...
+	. = ..()
+
+/obj/item/organ/stomache/process()
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
/obj/item/organ/stomach/process()
```

**Comment by dragoslava** [comment]
this was a bad idea in retrospect, gonna revisit this later

File: code/modules/organs/organ_internal.dm [hunk 16]
```diff
@@ -15,6 +15,40 @@
...
+	contents_check() // if your stomache is too full, lets have you vomit up some of that stuff
+
+/obj/item/organ/stomache/proc/contents_check()
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
/obj/item/organ/stomach/proc/contents_check()
```

File: code/modules/organs/organ_internal.dm [hunk 17]
```diff
@@ -15,6 +15,40 @@
...
+	var/mob/living/carbon/human/H = owner
+	if(!H) 
+		return
+	if(contents.len > 4)
+		addtimer(CALLBACK(src, .proc/cough_up), rand(100, 400))
+		to_chat(owner, "<span class='danger'>You dont feel too good....</span>")
+		vomit_timer = TRUE
+
+/obj/item/organ/stomache/proc/cough_up()
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
/obj/item/organ/stomach/proc/cough_up()
```

**Comment by VTCobaltblood** [comment]
**### V O R E S T A T I O N**

File: code/modules/organs/organ_internal.dm [hunk 18]
```diff
@@ -15,6 +15,40 @@
...
+/obj/item/organ/stomache/process()
+	contents_check() // if your stomache is too full, lets have you vomit up some of that stuff
+
+/obj/item/organ/stomache/proc/contents_check()
+	var/mob/living/carbon/human/H = owner
+	if(!H) 
+		return
+	if(contents.len > 4)
+		addtimer(CALLBACK(src, .proc/cough_up), rand(100, 400))
+		to_chat(owner, "<span class='danger'>You dont feel too good....</span>")
```
Comment (MarinaGryphon) [inline code/modules/organs/organ_internal.dm]:
```suggestion
		to_chat(owner, span("danger", "You don't feel too good..."))
```

File: html/changelogs/pooppr.yml [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes: 
+  - rscadd: "Adds the ability to swallow small items."
+  - rscadd: "Adds a new stomache organ to humans."
```
Comment (MarinaGryphon) [inline html/changelogs/pooppr.yml]:
```suggestion
  - rscadd: "Adds a new stomach organ to humans."
```

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 20]
```diff
@@ -270,6 +270,38 @@
...
+	set desc = "Swallows a held object"
+	set category = "IC"
+
+	var/mob/living/carbon/human/H = src
+	var/obj/item/organ/ST = H.internal_organs_by_name["stomache"]
+	var/obj/item/organ/external/E = H.organs_by_name[ST.parent_organ]
+	var/obj/item/I = H.get_active_hand()
+
+	if(!I)
+		to_chat(H, "<span class='notice'>You can't swallow nothing!</span>")
```
Comment (MarinaGryphon) [inline code/modules/mob/living/carbon/human/human_powers.dm]:
```suggestion
```

Comment (MarinaGryphon) [reply code/modules/mob/living/carbon/human/human_powers.dm]: Just have this be a silent failure.

**Comment by MarinaGryphon** [comment]
whoops, didn't see that it got closed

**Pull request closed without merging**